src/Atmos/Model/ref_state.jl

```diff
@@ -138,11 +147,10 @@ function (profile::DryAdiabaticProfile)(
     z = altitude(orientation, param_set, aux)
 
     # Temperature
-    T = profile.T_surface - _grav * z / _cp_d
+    T = max(profile.T_surface - _grav * z / _cp_d, profile.T_min)
 
     # Pressure
     p = _MSLP * (1 - _grav * z / (_cp_d * profile.T_surface))^(_cp_d / _R_d)
-
```

Comment:
You need to make the pressure consistent with the temperature profile (i.e., replace with the pressure profile that you had before that included the temperature cutoff.

Reply:
I thought this might have been needed and was going to ask before merging. Thanks!